keps/sig-storage/20190204-kubernetes-csi-release-process.md

```diff
@@ -0,0 +1,302 @@
+---
+title: kubernetes-csi release process
+authors:
+  - "@pohly"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-testing
+  - sig-release
+reviewers:
+  - @msau42
+approvers:
+  - @msau42
+  - sig-testing: TBD
+  - sig-release: TBD
+editor: @pohly
+creation-date: 2019-02-04
+last-updated: 2019-02-04
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# kubernetes-csi-release-process
+
+## Table of Contents
+
+- [kubernetes-csi-release-process](#kubernetes-csi-release-process)
+    - [Table of Contents](#table-of-contents)
+    - [Summary](#summary)
+    - [Motivation](#motivation)
+        - [Goals](#goals)
+        - [Non-Goals](#non-goals)
+    - [Proposal](#proposal)
+        - [Versionioning](#versionioning)
+            - [Release artifacts](#release-artifacts)
+            - [Release process](#release-process)
+        - [Implementation Details](#implementation-details)
+        - [Risks and Mitigations](#risks-and-mitigations)
+    - [Graduation Criteria](#graduation-criteria)
+    - [Implementation History](#implementation-history)
+    - [Drawbacks](#drawbacks)
+    - [Alternatives](#alternatives)
+    - [Infrastructure Needed](#infrastructure-needed)
+
+
+## Summary
+
+The Kubernetes [Storage
+SIG](https://github.com/kubernetes/community/tree/master/sig-storage)
+maintains a set of components under the
+[kubernetes-csi](https://github.com/kubernetes-csi) GitHub
+organization. Those components are intentionally not part of core
+Kubernetes even though they are maintained by the Kubernetes project.
+
+This document explains how these components are released.
+
+## Motivation
+
+So far, the process for tagging and building components has been
+fairly manual, with some assistance by Travis CI. There has been no
+automatic testing on a Kubernetes cluster, neither stand-alone
+deployment of individual components nor full end-to-end (E2E) testing
+of several Kubernetes-CSI components.
```

Comment:
Worth clarifying there is indirect E2E testing via the Kubernetes/Kubernetes tests?

Reply:
I can mention it. But the automatic testing was for already released images, not the pre-submit and pre-release testing that we are now trying to add.